pkg/provider/aws/expose.go

```diff
@@ -0,0 +1,123 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/klothoplatform/klotho/pkg/annotation"
+	"github.com/klothoplatform/klotho/pkg/core"
+	"github.com/klothoplatform/klotho/pkg/multierr"
+	"github.com/klothoplatform/klotho/pkg/provider/aws/resources"
+	"github.com/pkg/errors"
+)
+
+func (a *AWS) GenerateExposeResources(gateway *core.Gateway, result *core.ConstructGraph, dag *core.ResourceGraph) error {
+	err := a.CreateRestApi(gateway, result, dag)
+	return err
+}
+
+func (a *AWS) CreateRestApi(gateway *core.Gateway, result *core.ConstructGraph, dag *core.ResourceGraph) error {
+	var errs multierr.Error
+	api := resources.NewRestApi(a.Config.AppName, gateway)
+	dag.AddResource(api)
+	api_references := []core.AnnotationKey{gateway.AnnotationKey}
+	triggers := map[string]string{}
+
+	resourceMap := map[string]*resources.ApiResource{}
```

Comment:
minor-ish: with all the things going on below, it's hard to remember exactly what this map is. Maybe rename to `apisBySegment` or similar?

Reply:
yeah i can rename